### PR TITLE
[feature](Nereids): support comparing mv with inferred predicate

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

support comparing mv with inferred predicate

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/HyperGraphComparator.java [hunk 1]
```diff
@@ -108,16 +109,22 @@ private ComparisonResult isLogicCompatible() {
     private ComparisonResult buildComparisonRes() {
         ComparisonResult.Builder builder = new ComparisonResult.Builder();
         for (Entry<Edge, List<? extends Expression>> e : pullUpQueryExprWithEdge.entrySet()) {
-            if (!e.getValue().isEmpty() && !canPullUp(e.getKey())) {
+            List<? extends Expression> rawFilter = e.getValue().stream()
+                    .filter(expr -> !expr.isInferred())
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/HyperGraphComparator.java:113]: this should use method `org.apache.doris.nereids.util.ExpressionUtils#isInferred`

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by seawinde — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**